ycmd.el

```diff
@@ -551,7 +551,8 @@ and `delete-process'.")
     (after-change-functions           . ycmd--on-change)
     (window-configuration-change-hook . ycmd--on-window-configuration-change)
     (kill-buffer-hook                 . ycmd--teardown)
-    (before-revert-hook               . ycmd--teardown))
+    (before-revert-hook               . ycmd--teardown)
+    (hack-local-variables-hook        . ycmd--on-open))
```

Comment:
After I got in bed I started thinking the same thing :-). I was miming from `tide-mode`, which is the typescript-ide mode, and I think this method made more sense in that context. I'll make the change.